### PR TITLE
fix: Addition of default pad token in tokenizer when EOS and PAD token are equal

### PR DESCRIPTION
**Pull request by Abhishek-TAMU**
### Description of the change

For any model when tokenizer has PAD token equal to EOS token, then the `DataCollatorForCompletionOnlyLM` masks the EOS token from `labels` of dataset batch which leads to repetitive `predicted_target` during inference time thus decreasing f1-micro for models like `granite-34b-gptq` 

### Related issue number

[#1309 ](https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/1309)

### How to verify the PR

Running HF inference after QLoRA tuning of `granite-34b-gptq` model. 

### Was the PR tested

- [ ] I have added >=1 unit test(s) for every new method I have added.
- [x] I have ensured all unit tests pass

### CODE REVIEW

**Review by Ssukriti — APPROVED** [review]

**Comment by Ssukriti** [comment]
Thank you @Abhishek-TAMU . Lets wait for another review by @kmehant since he was closely involved with the tokens update.

File: tuning/sft_trainer.py
Comment (kmehant) [inline tuning/sft_trainer.py:267]: @Ssukriti @Abhishek-TAMU what if the eos_token in tokenizer is actually the DEFAULT_PAD_TOKEN? Can we handle that case?

Comment (kmehant) [reply tuning/sft_trainer.py:267]: example - 
tokenizer.eos_token is configs.DEFAULT_PAD_TOKEN
and 
tokenizer.pad_token is configs.DEFAULT_PAD_TOKEN

At this point, we are setting tokenizer.pad_token  to configs.DEFAULT_PAD_TOKEN this again. We might want to think about this case.

Comment (Abhishek-TAMU) [reply tuning/sft_trainer.py:267]: Good point. In this case can we modify and assign `eos_token` as `configs.DEFAULT_EOS_TOKEN` ? @Ssukriti

Comment (Ssukriti) [reply tuning/sft_trainer.py:267]: its very unlikely that EOS token will be <PAD> no?

File: tuning/sft_trainer.py
Comment (Ssukriti) [inline tuning/sft_trainer.py:266]:
```suggestion
            logger.warning("PAD token set to default, to make it different from eos token")
```

File: tuning/sft_trainer.py
Comment (Ssukriti) [reply tuning/sft_trainer.py:267]: but ya if we want to be absolutely sure, we can do that

if tokenizer.pad_token == tokenizer.eos_token:
     if tokenizer.eos_token != configs.DEFAULT_PAD_TOKEN:
            tokenizer.pad_token  = configs.DEFAULT_PAD_TOKEN
     else:
           tokenizer.eos_token = configs.DEFAULT_EOS_TOKEN

**Comment by Abhishek-TAMU** [comment]
Pushed the suggested changes. Thank you!
@kmehant @Ssukriti

**Review by kmehant — APPROVED** [review]

**Pull request merged**
